server/api/recipes.js

```diff
@@ -16,15 +16,46 @@ router.get('/:id', (req, res, next) => {
 });
 
 router.post('/', (req, res, next) => {
-  Recipe.create(req.body)
-    .then((newRecipe) => {
-      res.status(201).json(newRecipe);
+  Recipe.findOrCreate({
+    where: {
+      title: req.body.title,
+      author: req.body.author,
+    },
+    defaults: {
+      recipeUrl: req.body.recipeUrl,
+      imageUrl: req.body.imageUrl,
+      siteName: req.body.siteName,
+      tags: req.body.tags,
+      isFavorite: req.body.isFavorite,
+      numServings: req.body.numServings,
+    },
+  })
+    .spread((newRecipe, isCreated) => {
+      if (isCreated) {
+        const arrIngredients = req.body.ingredients.split(',');
+        const arrObjIngredients = [];
+        arrIngredients.forEach((ingredient) => {
```

Comment:
change to map and only do the ingredient create

so then you'll have an array of ing promises